File: ios/LiveActivityModule.m
//
//  LiveActivityModule.m
//  go2go_mobile
//
//  Created by Leekayoung on 6/17/25.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(LiveActivity, NSObject)

RCT_EXTERN_METHOD(startActivity)
RCT_EXTERN_METHOD(endActivity)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}
  
@end

File: ios/LiveActivityModule.swift
import SwiftUI
import ActivityKit


@objc(LiveActivity)
class LiveActivityModule: NSObject {
  
  private var content: ActivityContent<TimerLiveActivityAttributes.ContentState>?

  @objc(startActivity)
  func startActivity() {
    do {
      if #available(iOS 16.1, *) {
        let liveActAttributes = TimerLiveActivityAttributes(name: "test")
        let liveActContentState = TimerLiveActivityAttributes.ContentState(emoji: "")
        content = ActivityContent(state: liveActContentState, staleDate: nil, relevanceScore: 1.0)
        if let content{
          try Activity.request(attributes: liveActAttributes , content: content, pushType: nil)
        }
      }
    } catch {
      print("Error")
    }
  }

  @objc(endActivity)
  func endActivity() {
    Task {
      for activity in Activity<TimerLiveActivityAttributes>.activities {
        await activity.end(content, dismissalPolicy: .default)
      }
    }
  }
}
